src/v2/controllers/vdeployment_controller/trusted/rely_guarantee.rs

```diff
@@ -0,0 +1,159 @@
+                let owner_references = etcd_obj.metadata.owner_references.get_Some_0();
+                &&& s.resources().contains_key(req.key)
+                &&& etcd_obj.metadata.resource_version.is_Some()
+                // &&& etcd_obj.metadata.resource_version
+                //     == req.preconditions.get_Some_0().resource_version
+                &&& etcd_obj.metadata.owner_references.is_Some()
+                // Can pass the owner_references check
+                &&& owner_references.contains(req.owner_ref)
+                &&& exists |vrs: VReplicaSetView| 
+                    #[trigger] req.owner_ref == vrs.controller_owner_ref()
+            }
+    }
+}
+
+pub open spec fn vd_rely(other_id: int) -> StatePred<ClusterState> {
+    |s: ClusterState| {
+        forall |msg| {
+            &&& #[trigger] s.in_flight().contains(msg)
+            &&& msg.content.is_APIRequest()
+            &&& msg.src == HostId::Controller(other_id)
+        } ==> match msg.content.get_APIRequest_0() {
+            APIRequest::CreateRequest(req) => vd_rely_create_req(req)(s),
+            APIRequest::GetThenUpdateRequest(req) => vd_rely_get_then_update_req(req)(s),
+            APIRequest::UpdateStatusRequest(req) => vd_rely_update_status_req(req)(s),
+            APIRequest::GetThenDeleteRequest(req) => vd_rely_get_then_delete_req(req)(s),
+            _ => true,
+        }
+    }
+}
+
+// VD Guarantee Condition
+
+// VD only creates VRS owned by itself.
+pub open spec fn vd_guarantee_create_req(req: CreateRequest) -> StatePred<ClusterState> {
+    |s: ClusterState| {
+        &&& req.obj.kind == Kind::CustomResourceKind("vreplicaset"@)
+        &&& req.obj.metadata.owner_references.is_Some()
+        &&& exists |vd: VDeploymentView| 
+            req.obj.metadata.owner_references.get_Some_0() == seq![#[trigger] vd.controller_owner_ref()]
+    }
+}
+
+// VD only updates VRS owned by itself.
+pub open spec fn vd_guarantee_get_then_update_req(req: GetThenUpdateRequest) -> StatePred<ClusterState> {
+    |s: ClusterState| {
+        let etcd_obj = s.resources()[req.key()];
+        &&& req.obj.kind == Kind::CustomResourceKind("vreplicaset"@)
+        &&& req.obj.metadata.resource_version.is_Some()
+        &&& exists |vd: VDeploymentView|
+            req.owner_ref == vd.controller_owner_ref()
+            && req.obj.metadata.owner_references.get_Some_0() == seq![#[trigger] vd.controller_owner_ref()]
+    }
+}
+
+// TODO: support delete requests (not yet implemented in the controller)
+// This should be the same as vrs_guarantee_delete_req
+// plus vd will not delete new vrs (has same template as vd)
+//      and will not delete when the number of vrs is not greater than vd.Spec.RevisionHistoryLimit
+pub open spec fn vd_guarantee_get_then_delete_req(req: GetThenDeleteRequest) -> StatePred<ClusterState> {
+    |s: ClusterState| {
+        false
+    }
+}
+
+pub open spec fn vd_guarantee(other_id: int) -> StatePred<ClusterState> {
```

Comment:
This is a problem in `vrs_guarantee` too, but it should be `controller_id` rather than `other_id` (since we're describing _our_ messages).